src/util.cpp

```diff
@@ -574,7 +574,10 @@ const fs::path &GetDataDir(bool fNetSpecific)
     if (fNetSpecific)
         path /= BaseParams().DataDir();
 
-    fs::create_directories(path);
+    if (fs::create_directories(path)) {
+        // This is the first run, create wallets subdirectory too
+        fs::create_directories(path / "wallets");
+    }
```

Comment:
Here and above: We should never create the wallets directory if wallet support is disabled. Additionally, if we're going to default to a wallet subdir, we should create it when there is no wallet yet, regardless of whether the datadir exists or not (eg, if datadir exists, but wallet.dat does not, make and use the wallet subdir).

Reply:
I had the same reaction when I first saw this, but I don't think it's a problem. If wallet support is disabled, we'll just have an empty wallet directory in the datadir.

Reply:
I think it's fine to create an empty wallets directory, no matter if wallets are enabled or not (either at build or runtime). Aesthetically it would be better otherwise but practically it makes sense:

As the presence of this directory is used as signal where to put wallets, transitioning to it for an existing data directory is non-trivial. How would it even know if a legacy datadir has no wallets and you're creating the first one? That would be some nightmare heuristic logic...